### ISSUE TITLE
How do I disable for spotlight?

### ISSUE THREAD

**Issue by haveanicedavid** [godbout/kindaVim.docs#205]
I'm sure I'm missing something obvious here, but can't for the life of me figure out how to disable kV in spotlight search?

**Comment by godbout** [comment]
hmm. yeah. Spotlight is not an app. not even sure it has a bundle ID. will have to investigate.

**Comment by godbout** [comment]
fuck. thought maybe Spotlight is part of System Settings, but no. it just seems to be a floating panel from the OS itself. shit. more investigation needed.

**Comment by Josh-Piper** [comment]
In the same boat, I ended up binding the activation using a sequence instead of escape, that way I chose when to use KV

**Comment by godbout** [comment]
> In the same boat

interesting

> I ended up binding the activation using a sequence instead of escape, that way I chose when to use KV

i don't really get this tho. what do you mean. the fact that `esc` is taken by kV is the issue? you can choose what combination replaces the macOS `esc` then, in the Settings. you can use for example `cmd esc` and pressing this would hide Spotlight right away. or am i missing something?

<img width="683" alt="Screenshot 2023-11-10 at 11 56 22" src="https://github.com/godbout/kindaVim.docs/assets/121373/27c02125-9fdb-422a-8c4b-3fa8f0116139">

**Comment by Josh-Piper** [comment]
Yeah disregard my previous comment. I think I was imagining things haha.

**Comment by haveanicedavid** [comment]
I'd still prefer being able to disable so I can make my muscle memory happy, but silly me just realized I can close out of spotlight with the same trigger command (`cmd + space`) so it's not really a big issue.

Originally posted this after moving from raycast (where I have KV disabled) to Spotlight, and when I'd open and want to cancel a search, I'd have to tap escape twice the normal amount. Hitting the original toggle accomplishes the same, and having vim movement in spotlight bar is growing on me

That to say - if this isn't an easy fix, probably not something that needs to be prioritized 😄

**Comment by godbout** [comment]
> I'd still prefer being able to disable so I can make my muscle memory happy, but silly me just realized I can close out of spotlight with the same trigger command (`cmd + space`) so it's not really a big issue.

ok. but `double escape` is too much for you? or one of the other solution proposed in kV's Settings? (any other modifier with `escape`.) i'm personally super used to `double escape` to get out of kV's Normal Mode.

> having vim movement in spotlight bar is growing on me

yeah i was wondering about that. because for the search, AND for moving through the result items, kV is pretty useful i think.

> That to say - if this isn't an easy fix, probably not something that needs to be prioritized 😄

the issue is as usual that Apple allows themselves stuff out of the ordinary. so Spotlight is not an app per se. it seems to have a bundle ID tho, and if i can use that, then i can add a manual way to add bundle IDs. there's already a couple of apps that are hardcoded in kV because no sane personal will not want them in the `Off Family`, but for Spotlight i'm sure some people wanna have kV in there. so yeah, the solution that comes to mind now is to add some sort of "manual adding an app to a Family", where you need to type the bundle ID yourself. will keep you updated.

**Comment by haveanicedavid** [comment]
> ok. but double escape is too much for you? 

I use single escape to get out of kV normal mode and that works great, but what I'm referring to is situations like: You've activated spotlight, scrolled to a search result, and want to close out of it. kV doubles however many `esc`'s I'd normally need... so instead of 3, I need 6. 

It's mildly annoying, but not that big a deal, and again - just triggering the spotlight shortcut again accomplishes what I want 

> the solution that comes to mind now is to add some sort of "manual adding an app to a Family", where you need to type the bundle ID yourself. will keep you updated.

If these situations are rare, could you just a toggle for certain apps? Like `Use in Spotlight` within kv settings?

**Comment by godbout** [comment]
> I use single escape to get out of kV normal mode and that works great, but what I'm referring to is situations like: You've activated spotlight, scrolled to a search result, and want to close out of it. kV doubles however many `esc`'s I'd normally need... so instead of 3, I need 6.

OH. just tried and had no idea Spotlight was working like this. got you now. thanks!

> It's mildly annoying, but not that big a deal, and again - just triggering the spotlight shortcut again accomplishes what I want

yeah. even without kV you actually need a bunch of `esc` to get out of Spotlight 🤣️ also you can't use the macOS standard `ctrl n` and `ctrl p` to go up and down the list of results? like you HAVE to use `up` and `down`? unbelievable. wtf is Apple doing with their UX.

> > the solution that comes to mind now is to add some sort of "manual adding an app to a Family", where you need to type the bundle ID yourself. will keep you updated.
> 
> If these situations are rare, could you just a toggle for certain apps? Like `Use in Spotlight` within kv settings?

good point. could start with something like this first. will give you credit for that one if done hahahahahahahahah

**Comment by godbout** [comment]
so yeah confirmed:
1. Spotlight doesn't have any app coz Apple, so can't drag and drop on kV's Settings
2. Spotlight still has a bundle ID so we can add it to Families

now the key is how to implement this.

**Comment by godbout** [comment]
Spotlight is now its own app, so you can add it to the `Off Family` like any other app. it's located in `/System/Library/CoreServices/`. open the folder, drag&drop, be happy.

<img width="1920" alt="Screenshot 2025-01-10 at 19 01 16" src="https://github.com/user-attachments/assets/03dce1bf-95fb-47c5-98a7-55e19a491b83" />

will add it to The Wizard in next release.
